src/protocols/aeternity/libs/AeAccountHdWallet.ts

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable class-methods-use-this */
+import {
+  AccountBase,
+  sign,
+  messageToHash,
+  RpcRejectedByUserError,
+  unpackTx,
+  Encoded,
+  METHODS,
+  buildTx,
+  Tag,
+  decode,
+} from '@aeternity/aepp-sdk';
+import { Ref } from 'vue';
+import type { ITx } from '@/types';
+import { useAccounts } from '@/composables/accounts';
+import { usePermissions } from '@/composables/permissions';
+import { PROTOCOL_AETERNITY } from '@/constants';
+
+interface InternalOptions {
+  fromAccount?: Encoded.AccountAddress;
+}
+
+export class AeAccountHdWallet extends AccountBase {
+  override readonly address: Encoded.AccountAddress;
+
+  nodeNetworkId: Ref<string | undefined>;
+
+  constructor(nodeNetworkId: Ref<string | undefined>) {
+    super();
+    const { getLastActiveProtocolAccount } = useAccounts();
+    this.address = getLastActiveProtocolAccount(PROTOCOL_AETERNITY)!.address;
+    this.nodeNetworkId = nodeNetworkId;
+  }
+
+  override async signTransaction(
+    txBase64: Encoded.Transaction,
+    options: Parameters<AccountBase['signTransaction']>[1],
+  ): Promise<Encoded.Transaction> {
+    if (!this.nodeNetworkId.value) {
+      throw new Error('Not connected to any network');
+    } else if (options?.aeppOrigin) {
+      const { checkOrAskPermission } = usePermissions();
+      const tx = unpackTx(txBase64) as any as ITx;
+      const permissionGranted = await checkOrAskPermission(
+        options.aeppOrigin,
+        METHODS.sign,
+        { ...options, txBase64, tx },
+      );
+      if (!permissionGranted) {
+        throw new RpcRejectedByUserError('Rejected by user');
+      }
+    }
+
+    const encodedTx = decode(txBase64);
+    const signature = await this.sign(
+      Buffer.concat([
+        Buffer.from(this.nodeNetworkId.value),
+        Buffer.from(encodedTx),
+      ]),
+      options, // Mainly to pass the `fromAccount` property
+    );
+    // @ts-ignore
```

Comment:
I suggest to replace this with `// @ts-expect-error` and to specify why it is needed

Reply:
It turns out I can remove this :) So cool to see the code just fit as expected.